src/DynamicData/Cache/ObservableCacheEx.cs

```diff
@@ -6457,6 +6513,11 @@ void UpdateAction(IChangeSet<TObject, TKey> updates)
         });
     }
 
+    private static Func<TObject, TKey, TResult> AdaptSelector<TObject, TKey, TResult>(Func<TObject, TResult> other)
```

Comment:
This seems a little silly as a dedicated helper function, if it's only being used once.

Reply:
I was thinking we could use it everywhere we have to adapt `(obj, key) => func(obj)`.  It would make the overloads that just exist to adapt the selector easier to see, but might be overkill.

Reply:
Something like this, I'd leave a comment or a TODO specifying why the "normal" code was bypassed.

Reply:
Done.